### PR TITLE
Allow user to define the express instance

### PR DESCRIPTION
**Pull request by ilarijuu**
Allows developers to add their own middleware to express. For example it allows adding authentication for the dashboard.

### CODE REVIEW

File: src/Mozaik.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ var winston = require('winston');
 var path    = require('path');
 var chalk   = require('chalk');
 var Bus     = require('./Bus');
+var express     = require('express');
```
Comment (plouc) [inline src/Mozaik.js]: this should be aligned with previous vars

File: src/server.js [hunk 2]
```diff
@@ -26,7 +24,7 @@ module.exports = function (mozaik) {
 
     app.get('/', function (req, res) {
         res.render('index', {
-            env:           config.env,
+            env:           mozaik.config.env,
```
Comment (plouc) [inline src/server.js]: I think it should be moved to another PR as it's not related to custom express instance

File: src/server.js [hunk 3]
```diff
@@ -7,12 +7,10 @@ var _       = require('lodash');
 /**
  * @param {Mozaik} mozaik
```
Comment (plouc) [inline src/server.js]: Missing doc on `app`

**Comment by juhamust** [comment]
Is there something missing from the PR to getting it merged? The CI testing failures seems to arise from ws module when tried to install with node 0.10

**Comment by plouc** [comment]
@juhamust, it's OK, just merged it. @ilarijuu thanks for your contribution!

**Comment by juhamust** [comment]
:+1:
